package-res/resources/web/pentaho/type/simple.js

```diff
@@ -246,6 +246,27 @@ define([
       },
       //endregion
 
+      toSpec: function(keyArgs) {
```

Comment:
I think that allocating memory you might not need is undesirable in a method of this "granularity".